testsuite/src/main/scala/endpoints/testsuite/BasicAuthTestApi.scala

```diff
@@ -12,4 +13,13 @@ trait BasicAuthTestApi extends algebra.Endpoints with algebra.BasicAuthenticatio
     textResponse
   )
 
+  val protectedEndpointViaBuilder =
+    anEndpoint
+      .withMethod(Get)
+      .withUrl(path / "users")
+      .withTextResponse
+      .withBasicAuth
```

Comment:
If one writes:

~~~ scala
anEndpoint
  .withMethod(Get)
  .withUrl(path / "users")
  .withBasicAuth
  .withTextResponse
~~~

Then the behavior is unexpected: authentication will be handled only for requests, not responses (this is because `.withTextResponse` is called *after* `withBasicAuth`).

We could guard against these kind of incompatible combinations of builder operations by using more refined types, but I’m afraid that would be super complex :(